cdap-common/src/main/java/io/cdap/cdap/common/metadata/QueryTerm.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.common.metadata;
+
+/**
+ * Represents a single item in a search query in terms of its content (i.e. the value being searched for)
+ * and its qualifying information (e.g. whether a match for it is optional or required).
+ * Is typically constructed in a list via {@link QueryParser#parse(String)}
+ */
+public class QueryTerm {
+  private String term;
```

Comment:
Mark field as `final`.

Reply:
Addressed in https://github.com/cdapio/cdap/pull/11478